src/evt/player_api.c

```diff
@@ -518,84 +973,483 @@ ApiStatus func_802D354C(ScriptInstance* script, s32 isInitialCall) {
         return ApiStatus_DONE2;
     }
 
-    temp->unk_00 = ALT_load_entity_model(unkStructPtr);
-    temp->unk_04 = 0;
-    temp->unk_08 = 0;
-    temp->unk_0C = 0;
-    temp->unk_10 = 0;
-    temp->unk_14 = 0;
-    temp->unk_18 = 0;
-    temp->unk_1C = 1.0f;
-    temp->unk_20 = 1.0f;
-    temp->unk_24 = 1.0f;
-
-    exec_entity_model_commandlist(temp->unk_00);
+    virtualEntity->entityModelIndex = ALT_load_entity_model(unkStructPtr);
+    virtualEntity->pos.x = 0.0f;
+    virtualEntity->pos.y = 0.0f;
+    virtualEntity->pos.z = 0.0f;
+    virtualEntity->rot.x = 0.0f;
+    virtualEntity->rot.y = 0.0f;
+    virtualEntity->rot.z = 0.0f;
+    virtualEntity->scale.x = 1.0f;
+    virtualEntity->scale.y = 1.0f;
+    virtualEntity->scale.z = 1.0f;
+
+    exec_entity_model_commandlist(virtualEntity->entityModelIndex);
     set_variable(script, outVar, i);
 
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(ApiStatus, "evt/player_api", func_802D3624, ScriptInstance* script, s32 isInitialCall);
+ApiStatus func_802D3624(ScriptInstance* script, s32 isInitialCall) {
+    VirtualEntity* temp = (*D_802DB7C0)[get_variable(script, *script->ptrReadPos)];
```

Comment:
maybe a more descriptive name for these? E.g. above, "virtualEntity" is used. Here and below if you decide to do it